src/core/client_channel/subchannel.cc

```diff
@@ -966,6 +1055,29 @@ bool Subchannel::PublishTransportLocked() {
     connected_subchannel_ = MakeRefCounted<NewConnectedSubchannel>(
         std::move(*call_destination), std::move(transport_destination), args_);
   }
+
+  // Extract security level from auth context
```

Comment:
I suggest moving this code into the `ConnectedSubchannel` ctor, since that's where we're storing the value.